sfa_dash/plotting/timeseries.py

```diff
@@ -0,0 +1,119 @@
+from bokeh.embed import components
+from bokeh.models import ColumnDataSource
+from bokeh.plotting import figure
+import pandas as pd
+
+from sfa_dash.filters import api_varname_to_units, api_to_dash_varname
+
+
+def build_df(json):
+    """Parses a Solar Forecast Arbiter API JSON
+    response into a pandas dataframe
+
+    Parameters
+    ----------
+    json: dict
+        The parsed API response from a /values endpoints.
+
+    Returns
+    -------
+    Dataframe
+        A dataframe parsed from the response's 'values' field.
+    """
+    values = json.get('values')
+    if not values:
+        raise ValueError('Empty Values field')
+    df = pd.DataFrame(values)
+    df = df.set_index('timestamp')
+    df.index = pd.to_datetime(df.index)
+    return df
+
+
+def build_x_axis_label(metadata):
+    """Builds an x axis label from a metadata object.
+
+    Parameters
+    ----------
+    metadata:
+        Metadata dictionary for the resource. Must include the "variable"
+        key.
+
+    Returns
+    string
+        The appropriate x axis label.
+    """
+    variable = metadata['variable']
+    var_name = api_to_dash_varname(variable)
+    units = api_varname_to_units(variable)
+    label = f'{var_name}{units}'
+    return label
+
+
+def build_figure_title(metadata, start, end):
+    """Builds a title for the plot from a metadata object.
+
+    Parameters
+    ----------
+    metadata: dict
+        Metadata dictionary used to label the plot. Must include a 'site'
+        key containing a nested site object as well as the 'variable' key.
+
+    start: datetime-like
+        The start of the interval being plot.
+
+    end: datetime-like
+        The end of the interval being plot.
+
+    Returns
+    -------
+    string
+        The appropriate figure title.
+    """
+    site_name = metadata['site']['name']
+    start_string = start.strftime('%Y-%m-%d %H:%M')
+    end_string = end.strftime('%Y-%m-%d %H:%M')
+    figure_title = (f'{site_name} {build_x_axis_label(metadata)}'
```

Comment:
I think this should be the observation name and we probably don't need the variable/units there so ``f'{observation_name} from {start_string} to {end_string}'``